scripts/o3de/tests/ExportScripts/test_export_source_built_project.py

```diff
@@ -0,0 +1,1296 @@
+         patch('o3de.export_project.kill_existing_processes') as mock_kill_processes,\
+         patch('o3de.export_project.build_export_toolchain') as mock_build_export_toolchain,\
+         patch('o3de.export_project.build_game_targets') as mock_build_game_targets,\
+         patch('o3de.export_project.get_asset_processor_batch_path') as mock_get_asset_processor_path,\
+         patch('o3de.export_project.build_assets') as mock_build_assets,\
+         patch('o3de.export_project.get_asset_bundler_batch_path') as mock_get_asset_bundler_path,\
+         patch('o3de.export_project.bundle_assets') as mock_bundle_assets,\
+         patch('o3de.export_project.setup_launcher_layout_directory') as mock_setup_launcher_layout_directory:
+        
+        mock_ctx = create_autospec(O3DEScriptExportContext)
+        mock_ctx.project_path = test_project_path
+        mock_ctx.engine_path = test_engine_path
+        mock_ctx.project_name = test_project_name
+
+        mock_platform = 'pc'
+
+        for should_build_tools, toolconf in zip([True, False, True, False],
+                                                ["profile", "profile", "release", "release"]):
+            for base_path in [None, test_o3de_base_path, test_absolute_base_path, test_relative_base_path]:
+
+                test_tools_build_path = None if not base_path else (base_path / "build" / "tools") 
+                test_asset_bundling_path = None if not base_path else (base_path / 'build' / 'asset_bundling')
+
+
+                test_tools_sdk_path = (test_engine_path / 'bin/Windows' / toolconf / 'Default')
+                
+
+                if test_tools_build_path and test_tools_build_path.is_absolute():
+                    test_tools_build_path.mkdir(exist_ok=True, parents=True)
+                    
+
+                export_standalone_project(mock_ctx,
+                                        mock_platform,
+                                        test_output_path,
+                                        should_build_tools, 
+                                        "release",
+                                        toolconf,
+                                        [],[],[],
+                                        monolithic_build=use_monolithic,
+                                        engine_centric=is_engine_centric,
+                                        tools_build_path=test_tools_build_path,
+                                        asset_bundling_path=test_asset_bundling_path)
+                
+
+                cannot_build_monolithic = (use_sdk and not has_monolithic and use_monolithic)
+
+                if cannot_build_monolithic:
+                    mock_get_asset_bundler_path.assert_not_called()
+                    mock_bundle_assets.assert_not_called()
+                else:
+
+
+                    selected_tools_build_path = test_tools_build_path if not use_sdk else test_tools_sdk_path
```

Comment:
nitpick: extra newlines
```suggestion
                else:
                    selected_tools_build_path = test_tools_build_path if not use_sdk else test_tools_sdk_path
```